src/main/java/subway/line/domain/SectionsInAllLine.java

```diff
@@ -0,0 +1,37 @@
+package subway.line.domain;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import subway.line.exception.InvalidStationIdException;
+import subway.station.domain.Station;
+
+public class SectionsInAllLine {
+  private final List<SectionWithFare> sections;
+  private final Map<Long, Station> stationMap;
+
+  public SectionsInAllLine(List<SectionWithFare> sections) {
+    this.sections = sections;
+    this.stationMap = createStationMap();
+  }
+
+  public List<SectionWithFare> getSections() {
+    return sections;
+  }
+
+  private Map<Long, Station> createStationMap() {
+    return sections.stream()
+        .flatMap(section -> section.getStations().stream())
```

Comment:
```suggestion
        .flatMap(section -> Stream.of(section.getUpStation(), section.getDownStation()))
```
이렇게 하면 `Section.getStations`를 없앨 수도 있습니다

Reply:
Stream.of()를 활용하면 Section.getStations를 삭제할 수 있겠네요. 수정했습니다.